pkg/utils/retry.go

```diff
@@ -47,13 +47,11 @@ func WithRetry(
 	return allErrors // nolint:wrapcheck
 }
 
-// MessageIsRetryableS3Error checks whether the message returning from TiKV is retryable ExternalStorageError.
-func MessageIsRetryableS3Error(msg string) bool {
+// MessageIsRetryableStorageError checks whether the message returning from TiKV is retryable ExternalStorageError.
+func MessageIsRetryableStorageError(msg string) bool {
 	msgLower := strings.ToLower(msg)
-	// If failed to read/write to S3.
-	failed := strings.Contains(msgLower, "failed to put object") || strings.Contains(msgLower, "failed to get object")
-	// If s3 stop or not start.
-	closedOrRefused := strings.Contains(msgLower, "server closed") || strings.Contains(msgLower, "connection refused")
-	// Those conditions are retryable.
-	return failed && closedOrRefused
+	// UNSAFE! TODO: Add a error type for retryable connection error.
+	// If S3/GCS stop or not start.
+	return strings.Contains(msgLower, "server closed") || strings.Contains(msgLower, "writing a body to connection") ||
```

Comment:
Maybe we can use an array here and test all GCS error messages here.

BTW, maybe `"error during dispatch"` would be a more generic pattern? 🤔